app/controllers/auth/sessions_controller.rb

```diff
@@ -35,10 +35,13 @@ def user_params
     params.require(:user).permit(:email, :password, :otp_attempt)
   end
 
-  def after_sign_in_path_for(_resource)
+  def after_sign_in_path_for(resource)
     last_url = stored_location_for(:user)
 
-    if [about_path].include?(last_url)
+    if [about_path].include?(last_url) ||
+       (single_user_mode? &&
+        resource.is_a?(User) &&
+        last_url == short_account_path(username: resource.account))
```

Comment:
OK.  Do you want the whole condition set to be factored out or just the part after `||` which I added?